File: application/src/Caravelo/Modules/Survey/Domain/Model/Question.php
<?php

namespace Caravelo\Modules\Survey\Domain\Model;


class Question
{
    /** @var int $id */
    private $id;

    /** @var  string $title */
    private $title;

    /** @var  Answer[] $answers */
    private $answers = [];

    /** @var  \DateTime $createdAt */
    private $createdAt;

    /**
     * Question constructor.
     * @param int $id
     * @param string $title
     * @param \DateTime $createdAt
     */
    public function __construct($id, $title, \DateTime $createdAt)
    {
        $this->id = $id;
        $this->title = $title;
        $this->createdAt = $createdAt;
    }

    /**
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @return string
     */
    public function getTitle()
    {
        return $this->title;
    }

    /**
     * @return Answer[]
     */
    public function getAnswers()
    {
        return $this->answers;
    }

    /**
     * @return \DateTime
     */
    public function getCreatedAt()
    {
        return $this->createdAt;
    }

    public function addAnswer(Answer $answer)
    {
        $this->answers[] = $answer;
    }
}

File: application-java/src/main/java/iruiz/survey/domain/model/Question.java
package iruiz.survey.domain.model;

import java.util.ArrayList;
import iruiz.utils.Slugify;

public final class Question {
	
	private String name;
	private String slug;
	private String description;
	private String createdAt;
	private ArrayList<Answer> answers;
	
	public Question(String name, String description, String createdAt){
		this.name = name;
		this.slug = Slugify.toSlug(name);
		this.description = description;
		this.createdAt = createdAt;
		this.answers = new ArrayList<Answer>();
	}
	
	public void addAnswer(Answer answer){
		this.answers.add(answer);
	}
	
	public ArrayList<Answer> getAnswers(){
		return this.answers;
	}
	
	public String getName(){
		return this.name;
	}
	
	public String getSlug(){
		return this.slug;
	}
	
	public String getDescription(){
		return this.description;
	}
	
	public String getCreatedAt(){
		return this.createdAt;
	}
}
